vars/runTest.groovy

```diff
@@ -117,33 +117,33 @@ def call(Map config = [:]) {
     // https://issues.jenkins-ci.org/browse/JENKINS-39203
     // Once that is fixed all of the below should be pushed up into the
     // Jenkinsfile post { stable/unstable/failure/etc. }
-    def status = "SUCCESS"
+    String status = 'SUCCESS'
     if (rc != 0) {
-        status = "FAILURE"
+        status = 'FAILURE'
     } else if (rc == 0) {
-        def test_failure = false
-        def test_error = false
+        boolean test_failure = false
+        boolean test_error = false
         if (config['junit_files']) {
-            def filesList = []
+            List filesList = []
             config['junit_files'].split().each {
                 filesList.addAll(findFiles(glob: it))
             }
             if (filesList) {
                 String junit_xml = filesList.collect{"'" + it + "'"}.join(' ')
                 if (sh(label: 'Check junit xml files for errors',
-                       script: 'grep "<error " ' + junit_xml,
+                       script: 'grep "<error.* " ' + junit_xml,
                        returnStatus: true) == 0) {
-                    status = "FAILURE"
-                    println "Found at least one error in the Junit files."
+                    status = 'FAILURE'
+                    println 'Found at least one error in the Junit files.'
                 } else if (sh(label: 'Check junit xml files for failures',
-                              script: 'grep "<failure " ' + junit_xml,
+                              script: 'grep "<failure.* " ' + junit_xml,
```

Comment:
I'm a little worried that this pattern is overly broad.  Maybe two different more specific patterns using `grep -e` is safer.

Reply:
We do not have a specification for JUnit that we can reference to determine what is safer.
`grep -E ".*<error( |Details>|StackTrace>)" junit.xml` is more precise in the matching what was missed.
Not finding any examples of equivalents suffixes for "failure".